lib/websocket/receiver.js

```diff
@@ -254,6 +233,43 @@ class ByteParser extends Writable {
     }
   }
 
+  /**
+   * Take n bytes from the buffered Buffers
+   * @param {number} n
+   * @returns {Buffer|null}
+   */
+  consume (n) {
+    if (n > this.#byteOffset) {
+      return null
+    } else if (n === 0) {
+      return Buffer.allocUnsafe(0)
+    }
+
+    const buffer = Buffer.allocUnsafe(n)
+    let offset = 0
+
+    while (offset !== n) {
+      const next = this.#buffers[0]
+      const { length } = next
+
+      if (length + offset === n) {
+        buffer.set(this.#buffers.shift(), offset)
```

Comment:
If you get here with `offset === 0` no copy and no `Buffer.allocUnsafe(n)` is needed, you can just return `this.#buffers.shift()`.